### DISCUSSION TITLE
Does anyone still use `textadept.editing.autocomplete_all_words`?

### DISCUSSION THREAD

**Discussion by orbitalquark** [orbitalquark/textadept#339] [General]
I'm considering removing this field unless there are objections. I believe it was a user request back in 2014, but I don't know if it's still useful now.

**Comment by orbitalquark** [comment]
For reference:

textadept.editing.autocomplete_all_words (bool)

Autocomplete the current word using words from all open buffers. If true, performance may be slow when many buffers are open. The default value is false.

**Comment by paaguti** [comment]
I sometimes activate it for medium sized (3/4 files <200 lines of Python
each) and it does a nice enough job to make me wonder whether I need LSP ;-)
And, additionally, if you leave it as false, there is no impact on
performance. Is it really worth killing this feature?

My .2 cents, /PA

On Sun, 15 Jan 2023 at 17:25, orbitalquark ***@***.***> wrote:

> For reference:
>
> textadept.editing.autocomplete_all_words (bool)
>
> Autocomplete the current word using words from all open buffers. If true,
> performance may be slow when many buffers are open. The default value is
> false.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/orbitalquark/textadept/discussions/339#discussioncomment-4690663>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AAZKU4GWPSOOKXAFVLQUXQ3WSQQIBANCNFSM6AAAAAAT36I53Q>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Fragen sind nicht da um beantwortet zu werden,
Fragen sind da um gestellt zu werden
Georg Kreisler

Headaches with a Juju log:
unit-basic-16: 09:17:36 WARNING juju.worker.uniter.operation we should run
a leader-deposed hook here, but we can't yet

**Comment by benjcal** [comment]
I keep it on when working on C. I’d leave the header open on a separate tab and use autocomplete from there. I find it very easy to use depending on the header file I have open, i.e. I can close one and open another header when I’m done doing something and ready to do something else.

the same might be possible with ctags but I haven’t bothered to learn/use it. For context I use quite a bit of single header libraries like stb

**Comment by eric1234** [comment]
[I use it](https://github.com/eric1234/.textadept/blob/main/modules/autocomplete.lua#L12). I often have some sort of function, structure, object, etc in one file that I want to use in another file. There is a good chance in that case that I am working in both buffers so the word auto-complete scan all buffers is nice.

That being said, [I already monkey-patch the word autocomplete function](https://github.com/eric1234/.textadept/blob/main/modules/autocomplete.lua#L98-L129) due to not liking words in comments showing up in my auto-complete. Therefore if you removed this feature I could keep it in my monkey-patch.

**Comment by orbitalquark** [comment]
Thanks for the feedback everyone. This field will remain. I agree that 
it is useful.
